pyogrio/core.py

```diff
@@ -140,6 +143,7 @@ def read_info(path, layer=None, encoding=None):
                 "features": <feature count>
             }
     """
+    path = vsi_path(str(path))
```

Comment:
I think the following line can remove the cast to string now:
`ogr_read_info(str(path)` => `ogr_read_info(path`